### PR TITLE
Add option `--init-rule-docs`

### LINKED ISSUE #79: Add option to create missing rule docs
`--init-rule-docs` to create any missing rule docs.

Fail if there are no rule docs to create so this option isn't left in place indefinitely.

This would only be useful for someone who hasn't created any docs for their rules yet. In that scenario, this would help save time in getting started with documentation.

The reason we need an explicit option for this (and we don't want to automatically create missing rule docs by default) is because the rule docs might live at a non-standard path and we just didn't detect them. If the plugin has a non-standard rule doc path, the user just needs to specify `--path-rule-doc`.

Follow-up to https://github.com/bmish/eslint-doc-generator/issues/78.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Fixes #79

### CODE REVIEW

File: lib/generator.ts [hunk 1]
```diff
@@ -162,9 +164,14 @@ export async function generate(path: string, options?: GenerateOptions) {
...
-        `Could not find rule doc: ${relative(getPluginRoot(path), pathToDoc)}`
-      );
+      if (!initRuleDocs) {
+        throw new Error(
+          `Could not find rule doc: ${relative(getPluginRoot(path), pathToDoc)}`
+        );
+      }
+
+      mkdirSync(dirname(pathToDoc), { recursive: true });
+      writeFileSync(pathToDoc, '', {});
```
Comment (bmish) [inline lib/generator.ts]: I believe we can omit the `{}` or use `'utf8'` as the final argument.

File: lib/cli.ts [hunk 2]
```diff
@@ -133,6 +134,13 @@ export async function run(
       })`,
       parseBoolean
     )
+    .option(
+      '--init-rule-docs [boolean]',
+      `(optional) Whether to generate rule docs files if they don't yet exist. (default: ${
```
Comment (bmish) [inline lib/cli.ts]:
```suggestion
      `(optional) Whether to create rule doc files if they don't yet exist. (default: ${
```

**Review by bmish — CHANGES_REQUESTED** [review]
Looking good. 

1. Can you update the list of options in the README?
2. There's one more requirement, but it can be a follow-up PR if needed. `Fail if there are no rule docs to create so this option isn't left in place indefinitely.`

File: README.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# eslint-doc-generator
+-# eslint-doc-generator
```
Comment (JoshuaKGoldberg) [inline README.md]:
```suggestion
# eslint-doc-generator
```

File: test/lib/generate/file-paths-test.ts [hunk 4]
```diff
@@ -44,8 +95,8 @@ describe('generate (file paths)', function () {
 
     it('throws an error', async function () {
       // Use join to handle both Windows and Unix paths.
```
Comment (bmish) [inline test/lib/generate/file-paths-test.ts]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -124,6 +124,7 @@ There's also an optional path argument if you need to point the CLI to an ESLint
 | `--config-emoji` | Custom emoji to use for a config. Format is `config-name,emoji`. Default emojis are provided for [common configs](./lib/emojis.ts). To remove a default emoji and rely on a [badge](#badge) instead, provide the config name without an emoji. Option can be repeated. |
 | `--ignore-config` | Config to ignore from being displayed. Often used for an `all` config. Option can be repeated. |
 | `--ignore-deprecated-rules` | Whether to ignore deprecated rules from being checked, displayed, or updated (default: `false`). |
+| `--ignore-deprecated-rules` | Whether to create rule doc files if they don't yet exist (default: `false`). |
```
Comment (bmish) [inline README.md]:
```suggestion
| `--init-rule-docs` | Whether to create rule doc files if they don't yet exist (default: `false`). |
```

**Comment by bmish** [comment]
I'm working on fixing the stuck tests.

**Comment by bmish** [comment]
Tests should be fixed but GitHub Actions have been queued for an hour. Might be a GitHub issue. Will merge and release this when the tests complete.

**Review by bmish — APPROVED** [review]
Thanks so much for this!

**Pull request merged**